[v17] Refactor `enroll-resources/agents`

ptgott opened the pull request:
Backport #54612 to branch/v17

ptgott committed:
Refactor `enroll-resources/agents`

Give the section a more reasonable information architecture. The section
includes two children, one of which is a subsection (Joining Services)
and another is an introductory overview.

Move `agents/introduction.mdx` to make it the new `agents/agents.mdx`
section index page. This page is a high-level introduction, so it makes
more sense as the section index. Also edit the page to:

- Remove an unnecessary list of links to sections for each agent
  service.
- Condense the architecture overview H2 to pave the way for making this
  page more of a section intro and less of an intro/conceptual hybrid
  guide.

Move "Join services" guides to make them direct descendents of the
`agents` directory. This way, the substantial content of the section can
have a higher directory level.

Extract the information on adding a service to an existing agent into a
full how-to guide. This makes the instructions more visible. Also
include example snippets within the guide.

ptgott committed:
Respond to mmcallister feedback

- Fix spelling.
- Soften the language about recommending the Terraform provider and
  Kubernetes operator so we don't need to justify a recommendation.
  Instead, let readers decide for themselves.

ptgott committed:
Respond to stevenGravy feedback

Clarify the need to create a token for an additional service you want to
run.

ptgott committed:
Fix redirects

taraspos reviewed (changes requested):
@ptgott docs deployment failed with:

```
[cause]: Error: You are trying to create client-side redirections to invalid paths.
                                    
                                    These paths are redirected to but do not exist:
                                    - /ver/18.xdocs/pages/enroll-resources/agents/
```

Seems like `/` is missing after `18.x` for some reason 🤔

taraspos commented on docs/config.json:
```diff
@@ -419,6 +419,56 @@
+    },
+    {
+      "source": "/enroll-resources/agents/join-services-to-your-cluster/join-services-to-your-cluster/",
+      "destination": "docs/pages/enroll-resources/agents/",
```
```suggestion
      "destination": "/docs/pages/enroll-resources/agents/",
```

ptgott committed:
Fix malformed redirect (#54842)

This change fixes a redirect with a malformed destination path and
source path. The source path needs to have a single instance of the
`join-services-to-your-cluster` segment, since this is a section index
page. The destination path erroneously uses the beginning of a local
file path, and needs to omit the base `/docs/` segment.
